configure.py

```diff
@@ -317,8 +317,12 @@ def binary(name):
         cflags.remove('-fno-rtti')  # Needed for above pedanticness.
     else:
         cflags += ['-O2', '-DNDEBUG']
-    if 'clang' in os.path.basename(CXX):
-        cflags += ['-fcolor-diagnostics']
+    try:
+        proc = subprocess.Popen([CXX, '--version'], stdout=subprocess.PIPE)
+        if 'clang' in proc.communicate()[0].decode('utf-8'):
+            cflags += ['-fcolor-diagnostics']
```

Comment:
I wonder if we should spell this `-fdiagnostics-color` (clang understands both spellings) and also add it for gcc 4.9+. Probably good enough to do that once someone submits a patch for gcc.